### PR TITLE
Load recent videos/streams from YouTube

### PR DESCRIPTION
**Pull request by davidfischer**
This is branched from https://github.com/sandiegopython/pythonsd-django/pull/113. Either merge this first or switch the base branch after merging it.

Fixes https://github.com/sandiegopython/pythonsd-django/issues/50.

## Screenshot
Screenshot 2023-01-31 at 21-31-49 Home · San Diego Python

### CODE REVIEW

File: pythonsd/views.py [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 import pytz
 import requests
+from defusedxml.minidom import parseString
```
Comment (davidfischer) [inline pythonsd/views.py]:
```suggestion
```

File: pythonsd/tests/__init__.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 import webtest
 
 from config import wsgi
+from ..views import RecentVideosView
```
Comment (kjaymiller) [inline pythonsd/tests/__init__.py:19]: I would combine these into a single import.
```suggestion
from ..views import RecentVideosView, UpcomingEventsView
```

File: pythonsd/tests/__init__.py [hunk 3]
```diff
@@ -16,6 +16,7 @@
...
 from ..views import UpcomingEventsView
```
Comment (kjaymiller) [inline pythonsd/tests/__init__.py:20]:
```suggestion
```

File: pythonsd/tests/__init__.py [hunk 4]
```diff
@@ -131,6 +132,88 @@ def test_api_error(self):
...
+            self.assertContains(response, "Check out our")
+
+    def test_preloaded_videos(self):
+        with mock.patch(
+            "pythonsd.views.RecentVideosView.get_recent_videos",
+            return_value=self.expected_videos,
+        ) as mock_get:
+            response = self.client.get(self.url)
+            self.assertContains(response, "MvpiCyPpAhM")
+            self.assertContains(response, "<iframe")
```
Comment (kjaymiller) [inline pythonsd/tests/__init__.py:177]: I see that the widget is also being tested in `test_html_video_widget`. Is there a reason to check for the iframe twice.

**Review by kjaymiller — APPROVED** [review]
Looks good however there were questions on #113 that needed to be addressed, first. 

Also interesting choice on using requests + defusedxml...

I added a minor suggestion on the import in your tests (combining two to one line).

Again, great work!

**Comment by kjaymiller** [comment]
I see that @jlgimeno approved the last PR. This one should be good then, unless you wanted to make that minor change.

File: pythonsd/tests/__init__.py (hunk 2 above)
Comment (davidfischer) [reply pythonsd/tests/__init__.py:19]: I actually prefer one import per line as it makes diffs simpler. Usually I use [reorder python imports](https://pypi.org/project/reorder-python-imports/) to enforce it.

**Comment by davidfischer** [comment]
> Also interesting choice on using requests + defusedxml...

Defusedxml is the main only way to parse xml safely in python.

File: pythonsd/tests/__init__.py (hunk 4 above)
Comment (davidfischer) [reply pythonsd/tests/__init__.py:177]: There isn't but it is a double check in case we change things

**Pull request merged**